packages/iTwinUI-react/src/core/Carousel/CarouselSlider.tsx

```diff
@@ -46,53 +45,52 @@ export const CarouselSlider = React.forwardRef<
     setSlideCount(items.length);
   }, [items.length, setSlideCount]);
 
-  const [width, setWidth] = React.useState<number>();
-  const [resizeRef] = useResizeObserver(({ width }) => setWidth(width));
-
   const sliderRef = React.useRef<HTMLOListElement>(null);
-  const refs = useMergedRefs(sliderRef, resizeRef, ref);
-  const justMounted = React.useRef(true);
+  const refs = useMergedRefs(sliderRef, ref);
 
-  const previousWidth = React.useRef<number>();
-  React.useLayoutEffect(() => {
-    const slideToShow = sliderRef.current?.children.item(currentIndex) as
+  scrollToSlide.current = (
+    slideIndex: number,
+    { instant }: { instant?: boolean } = {},
+  ) => {
+    isManuallyUpdating.current = true; // start manual update
+
+    const slideToShow = sliderRef.current?.children.item(slideIndex) as
       | HTMLElement
       | undefined;
 
-    if (
-      !sliderRef.current ||
-      !slideToShow ||
-      (!isManuallyUpdating.current && previousWidth.current === width)
-    ) {
+    if (!sliderRef.current || !slideToShow) {
       return;
     }
 
-    // instant scroll on first mount
-    if (justMounted.current) {
-      scrollInstantly.current = true;
-      justMounted.current = false;
-    }
-
     const motionOk = getWindow()?.matchMedia(
       '(prefers-reduced-motion: no-preference)',
     )?.matches;
 
     sliderRef.current.scrollTo({
       left: slideToShow.offsetLeft - sliderRef.current.offsetLeft,
-      behavior: (scrollInstantly.current || !motionOk
-        ? 'instant'
-        : 'smooth') as ScrollBehavior, // scrollTo accepts 'instant' but ScrollBehavior type is wrong
+      behavior: (instant || !motionOk ? 'instant' : 'smooth') as ScrollBehavior, // scrollTo accepts 'instant' but ScrollBehavior type is wrong
     });
+  };
+
+  const scrollTimeout = React.useRef<number>();
+
+  // reset isManuallyUpdating.current to false after the last scroll event
+  const handleOnScroll = React.useCallback(() => {
+    if (scrollTimeout.current) {
+      clearTimeout(scrollTimeout.current);
+    }
 
-    scrollInstantly.current = false;
-    previousWidth.current = width;
-  }, [currentIndex, isManuallyUpdating, scrollInstantly, width]);
+    scrollTimeout.current = window.setTimeout(() => {
+      isManuallyUpdating.current = false;
+    }, 100);
```

Comment:
This is the only part that I'm not sure about. I had to do this because there is no `scrollend` event. To make things worse, React doesn't support `passive` scroll events, which is the recommended approach.

But at least the delay doesn't cause any usability problems (in my testing).

Reply:
Wouldnt `IntersectionObserver` help here? You could know, when element is fully visible (so assume scoll has ended)